pkg/kube/client/mocked_client.go

```diff
@@ -96,7 +100,50 @@ func (m *mockedClient) Update(_ context.Context, obj k8sClient.Object, _ ...k8sC
 	return nil
 }
 
-func (m *mockedClient) Patch(_ context.Context, _ k8sClient.Object, _ k8sClient.Patch, _ ...k8sClient.PatchOption) error {
+type patchValue struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value"`
+}
+
+func (m *mockedClient) Patch(_ context.Context, obj k8sClient.Object, patch k8sClient.Patch, _ ...k8sClient.PatchOption) error {
```

Comment:
For now we just implement mocked patch for what was needed now. 
Adding other options is outside the scope of this PR. I tried to cover every possible case and return a `not implemented` error when needed